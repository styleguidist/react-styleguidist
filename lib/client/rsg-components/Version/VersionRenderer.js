import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

var styles = function styles(_ref) {
  var color = _ref.color,
      fontFamily = _ref.fontFamily,
      fontSize = _ref.fontSize;
  return {
    version: {
      color: color.light,
      margin: [[5, 0, 0, 0]],
      fontFamily: fontFamily.base,
      fontSize: fontSize.base,
      fontWeight: 'normal'
    }
  };
};

export var VersionRenderer = function VersionRenderer(_ref2) {
  var classes = _ref2.classes,
      children = _ref2.children;
  return /*#__PURE__*/React.createElement("p", {
    "aria-label": "version",
    className: classes.version
  }, children);
};
VersionRenderer.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
  children: PropTypes.node
};
export default Styled(styles)(VersionRenderer);