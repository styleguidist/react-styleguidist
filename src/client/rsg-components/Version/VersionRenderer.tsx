import React from 'react';
import PropTypes from 'prop-types';
import Styled, { JssInjectedProps } from 'rsg-components/Styled';
import * as Rsg from '../../../typings';

const styles = ({ color, fontFamily, fontSize }: Rsg.Theme) => ({
	version: {
		color: color.light,
		margin: [[5, 0, 0, 0]],
		fontFamily: fontFamily.base,
		fontSize: fontSize.base,
		fontWeight: 'normal',
	},
});

interface VersionProps extends JssInjectedProps {
	children?: React.ReactNode;
}

export const VersionRenderer: React.FunctionComponent<VersionProps> = ({ classes, children }) => {
	return (
		<p aria-label="version" className={classes.version}>
			{children}
		</p>
	);
};

VersionRenderer.propTypes = {
	classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
	children: PropTypes.any,
};

export default Styled<VersionProps>(styles)(VersionRenderer);
