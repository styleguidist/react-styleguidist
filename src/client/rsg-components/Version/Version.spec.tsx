import React from 'react';
import renderer from 'react-test-renderer';
import VersionRenderer from './VersionRenderer';

it('renderer should render version', () => {
	const actual = renderer.create(<VersionRenderer>1.2.3-a</VersionRenderer>);

	expect(actual.toJSON()).toMatchSnapshot();
});
