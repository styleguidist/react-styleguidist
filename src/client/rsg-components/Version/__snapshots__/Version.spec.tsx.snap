// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render version 1`] = `
<p
  aria-label="version"
  className="rsg--version-0"
>
  1.2.3-a
</p>
`;
